src/strategies/ocean-marketplace-v4/index.ts

```diff
@@ -0,0 +1,175 @@
+      datatoken: {
+        holderCount: true
+      }
+    }
+  };
+
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.pools.__args.block = { number: +snapshot };
+  }
+
+  // Retrieve the top 1000 pools
+  const graphResults = await subgraphRequest(
+    OCEAN_SUBGRAPH_URL[network],
+    params
+  );
+
+  // Get total votes, for ALL addresses, inside top 1000 pools, with a minimum of 0.0001 shares
+  const score = {};
+  const userAddresses: string[] = [];
+  const return_score = {};
+
+  // console.log(
+  //   `graph results for network: ${network} at snapshot: ${snapshot}`
+  // );
+  // console.log('results: ', graphResults);
+
+  if (graphResults && graphResults.pools) {
+    graphResults.pools.forEach((pool) => {
+      if (pool.isFinalized) {
+        pool.shares.map((share) => {
+          const userAddress = getAddress(share.user.id);
+          // const shares = share.shares;
+          // console.log(
+          //   `High Level - User address: ${userAddress} user poolShares: ${shares} baseTokenLiquidity: ${pool.baseTokenLiquidity} poolTotalShares: ${pool.totalShares}`
+          // );
+          if (!userAddresses.includes(userAddress))
+            userAddresses.push(userAddress);
+          if (!score[userAddress]) score[userAddress] = BigNumber.from(0);
+          const userShare =
+            share.shares * (pool.baseTokenLiquidity / pool.totalShares);
+          if (userShare > 0.0001) {
+            score[userAddress] = score[userAddress].add(
+              bdToBn(userShare.toString(), OCEAN_ERC20_DECIMALS)
+            );
+          }
+        });
+      }
+    });
+
+    // We then sum total votes, per user address
+    userAddresses.forEach((address) => {
+      const parsedSum = parseFloat(
+        formatUnits(score[address], OCEAN_ERC20_DECIMALS)
+      );
+      return_score[address] = parsedSum;
+
+      // console.log(`Score for address: ${address} is: ${return_score[address]}`);
+    });
+  }
+
+  // We then filter only the addresses expected
+  const results = Object.fromEntries(
+    Object.entries(return_score).filter(([k]) => addresses.indexOf(k) >= 0)
+  );
+
+  // Test validation: Update examples.json w/ expectedResults to reflect LPs @ blockHeight
+  // Success criteria: Address scores and length, must match expectedResults. Order not validated.
+  // From GRT's graphUtils.ts => verifyResults => Scores need to match expectedResults.
+  // npm run test --strategy=ocean-marketplace | grep -E 'SUCCESS|ERROR'
+  if (options.expectedResults) {
+    const expectedResults = {};
+    Object.keys(options.expectedResults.scores).forEach(function (key) {
+      expectedResults[key] = results[key];
+    });
+
+    verifyResults(
+      JSON.stringify(expectedResults),
+      JSON.stringify(options.expectedResults.scores),
+      'Scores'
+    );
+
+    verifyResultsLength(
+      Object.keys(expectedResults).length,
+      Object.keys(options.expectedResults.scores).length,
+      'Scores'
+    );
+  }
```

Comment:
Hi @idiom-bytes I think you can remove code related to this, examples.json is just to test with few addresses, in real proposal we may receive thousands of addresses and can pass all of them in expectedresults

Reply:
@ChaituVR I'm just reviewing your feedback. 

We added these parameters, into examples.json as a way to provide a testing framework that enables us to check that the vote calculations are working, each step of the way.

Also, this will never run unless `options.expectedResults` is passed in.
Meaning that when thousands of addresses come through the process, this is not executed.

I'm going to propose/request that you accept this submission, so we can keep our examples.json up-to-date, and provide an easy way for developers to validate the voting score remains accurate across the chains we support.

Reply:
I am also happy to follow your path forward. If so, could you provide a way to test/scaffold more easily from our end, on top of snapshot-strategies...? I should probably review the testing documentation...